templates/index.html

```diff
@@ -0,0 +1,111 @@
+<html>
+
+<head>
+    <title>Monitoring Accessible Charts</title>
+    <script src="https://d3js.org/d3.v5.min.js"></script>
+</head>
+
+<body>
+    <h2>Monitoring Accessible Charts</h2>
+
+    <div>
+        <select name="selectedMode" id="modeSelector"></select>
+    </div>
+    <svg width="600" height="400" id="chart"></svg>
+    
+    <script>
+        /**
+         *  Fromats points for graphing.
+         */
+        function formatPoints(n, points) {
+            n = d3.min([n, points.length]);
+            const values = Array(n);
+            const dates = Array(n);
+            const window = points.length / n;
+            const middle = Math.floor(window / 2);
+
+            const sum = (acc, cur) => acc + cur["value"]["doubleValue"];
+
+            for (i = 0; i < n; i++) {
+                values[i] = (points.slice(i * window, (i + 1) * window)).reduce(sum, 0) / window;
+                dates[i] = new Date(points.slice(i * window, (i + 1) * window)
+                [middle]["interval"]["startTime"]);
+            }
+
+            return [values, dates];
+        }
+
+        /**
+         * Draws the chart.
+         */
+        const drawChart = async () => {
+            const url = 'http://127.0.0.1:5000/';
+            const response = await fetch(url+"data/chart-001.json");
```

Comment:
Let's add some quick error handling logic here. If the fetch fails, can we show an error message to the user?